camel-idea-plugin/src/main/java/org/apache/camel/idea/inspection/AbstractCamelInspection.java

```diff
@@ -152,6 +149,46 @@ private void validateSimple(@NotNull PsiElement element, final @NotNull Problems
         }
     }
 
+    private void validateJSonPath(@NotNull PsiElement element, final @NotNull ProblemsHolder holder, @NotNull String text, boolean isOnTheFly) {
+        CamelCatalog catalogService = ServiceManager.getService(element.getProject(), CamelCatalogService.class).get();
+        CamelService camelService = ServiceManager.getService(element.getProject(), CamelService.class);
+
+        IElementType type = element.getNode().getElementType();
+        LOG.trace("Element " + element + " of type: " + type + " to inspect jsonpath: " + text);
+
+        // must have camel-json library
+        boolean jsonLib = camelService.containsLibrary("camel-jsonpath", false);
+        if (!jsonLib) {
+            return;
+        }
+
+        try {
+            // need to use the classloader that can load classes from the project
+            ClassLoader loader = camelService.getProjectClassloader();
+            if (loader != null) {
+                LanguageValidationResult result;
+                boolean predicate = getCamelIdeaUtils().isCameJSonPathExpressionUsedAsPredicate(element);
+                if (predicate) {
+                    LOG.debug("Inspecting jsonpath predicate: " + text);
+                    result = catalogService.validateLanguagePredicate(loader, "jsonpath", text);
+                } else {
+                    LOG.debug("Inspecting jsonpath expression: " + text);
+                    result = catalogService.validateLanguageExpression(loader, "jsonpath", text);
+                }
+                if (!result.isSuccess()) {
+                    // favor the short error message
+                    String msg = result.getShortError();
+                    if (msg == null) {
+                        msg = result.getError();
+                    }
+                    holder.registerProblem(element, msg);
+                }
+            }
```

Comment:
consider split this up in small methods, like "validateLanguage" ,"registerProblem" eg.